router/tcp.go

```diff
@@ -34,6 +34,21 @@ type TCPListener struct {
 	closed   bool
 }
 
+func (s *TCPListener) List() ([]*router.Route, error) {
```

Comment:
`pgDataStore.List` doesn't filter on http vs tcp routes, it returns all the routes. This was a hack to filter out routes by type, there is an equivalent method in `HTTPListener`. Without these, the API returns duplicate routes (which causes `TestAPIListRoutes` to fail).